test/integ/test_connection.py

```diff
@@ -1370,3 +1372,39 @@ def test_token_file_path(tmp_path, db_parameters):
     assert conn._token == fake_token
     conn = snowflake.connector.connect(**db_parameters, token_file_path=token_file_path)
     assert conn._token == fake_token
+
+
+@pytest.mark.skipolddriver
+@pytest.mark.skipif(not RUNNING_ON_GH, reason="no ocsp in the environment")
+def test_mock_non_existing_server(conn_cnx, caplog):
+    from snowflake.connector.cache import SFDictCache
+
+    # this patch helps give a non-existing oscp cache server url
+    def patch_extrac_tld(**kwargs):
+        return "nonexistingtopleveldomain"
+
+    # disabling local cache and pointing ocsp cache server to a non-existing url
+    # connection should still work as it will directly validate the certs against CA servers
+    with tempfile.NamedTemporaryFile() as tmp, caplog.at_level(logging.DEBUG):
+        with mock.patch(
+            "snowflake.connector.url_util.extract_top_level_domain_from_hostname",
+            patch_extrac_tld,
+        ):
```

Comment:
nit: this would be more concise
```suggestion
        with mock.patch(
            "snowflake.connector.url_util.extract_top_level_domain_from_hostname",
            return_value="nonexistingtopleveldomain",
        ):
```